oxidation/libparsec/crates/types/src/manifest.rs

```diff
@@ -63,18 +86,45 @@ macro_rules! impl_manifest_dump_load {
                 let obj = rmp_serde::from_slice::<Self>(&serialized)
                     .map_err(|_| DataError::Serialization)?;
 
-                if obj.author != *expected_author {
+                obj.verify(
+                    expected_author,
+                    expected_timestamp,
+                    expected_id,
+                    expected_version,
+                )?;
+                Ok(obj)
+            }
+
+            /// Verify the manifest against a set of expected values
+            pub fn verify(
+                &self,
+                expected_author: &DeviceID,
+                expected_timestamp: DateTime,
+                expected_id: Option<EntryID>,
+                expected_version: Option<u32>,
+            ) -> Result<&Self, DataError> {
```

Comment:
Why returning `&Self` instead of `()` ?

(From what I've seen we always do `obj.verify([...])?;` given the verify always occurs just after object creation, so we always want the ownership on the object to return it to the our caller)

Reply:
Because the block that verify the values was in `verify_and_load` and was returning `Self`, but now that block is in `verify` method.

I will change the returned value